charts/seed-bootstrap/charts/monitoring/templates/config.yaml

```diff
@@ -101,3 +101,23 @@ data:
     - job_name: prometheus
       static_configs:
       - targets: [ localhost:9090 ]
+
+    - job_name: vpa-recommender
+      kubernetes_sd_configs:
+        - role: pod
+          relabel_configs:
+            - source_labels: [__meta_kubernetes_pod_label_app]
+              regex: ^vpa-recommender$
+              action: keep
+            - source_labels: [ __address__ ]
```

Comment:
```
      - source_labels: [ __meta_kubernetes_service_port_name ]
        regex: metrics
        action: keep
```
the pod service discovery discovers the pod-port combinations, so instead of hard coding a port, we should filter for the port we are interested in.

This assumes though that the pod's port is exposed in the pod manifest. Currently, it isn't.

```
$ k get deployment vpa-recommender -o json | gron | grep ports
json.spec.template.spec.containers[0].ports = [];
json.spec.template.spec.containers[0].ports[0] = {};
json.spec.template.spec.containers[0].ports[0].containerPort = 8080;
json.spec.template.spec.containers[0].ports[0].protocol = "TCP";
```

Could you add the port 8942 to the deployment manifest?

Reply:
done.

Reply:
Note that the example `__meta_kubernetes_service_port_name` didn't apply here, because we are using pod discovery.